src/Rewards.sol

```diff
@@ -208,19 +218,36 @@ contract Rewards is AccessControl, EIP712 {
         _mustBeFromAuthorizedOracle(digestBatchReward(batch), signature);
 
         _checkedUpdateQuota();
+
         uint256 batchSum = _batchSum(batch);
-        _checkedUpdateClaimed(batchSum);
+        uint256 submitterRewardAmount = (batchSum * batchSubmitterRewardPercentage) / 100;
+
+        _checkedUpdateClaimed(batchSum + submitterRewardAmount);
 
         // Safe to increment the sequence after checking this is the expected number (no overflow for the age of universe even with 1000 reward claims per second)
         batchSequence = batch.sequence + 1;
 
         for (uint256 i = 0; i < batch.recipients.length; i++) {
             nodl.mint(batch.recipients[i], batch.amounts[i]);
         }
+        nodl.mint(msg.sender, submitterRewardAmount);
 
         emit BatchMinted(batchSum, claimed);
     }
```

Comment:
I was actually under the impression the 2% would be a "tax" instead of an additional mint on top of the existing rewards; or were you thinking of handing this on the oracle side?

Reply:
This additional 2% is included as part of maximum daily mint, so it's not going to impact our total issuance. However I thought it's best to not alter what is already calculated and indicated by the oracle per individuals. Not only we would save gas this way, we keep these transactions simple and understandable. For example after a batch is issued, if the gov changes the reward percentage the rewards of the individuals in that unprocessed batch remains the same. On the oracle side we can be smarter and implement more sophisticated logic if it's ever required.